sklearn/utils/validation.py

```diff
@@ -498,10 +498,10 @@ def check_X_y(X, y, accept_sparse=None, dtype="numeric", order=None, copy=False,
                     ensure_2d, allow_nd, ensure_min_samples,
                     ensure_min_features, warn_on_dtype, estimator)
     if multi_output:
-        y = check_array(y, 'csr', force_all_finite=True, ensure_2d=False,
-                        dtype=None)
+        y = check_array(y, 'csr', copy=copy, force_all_finite=True,
+                        ensure_2d=False, dtype=None)
```

Comment:
Actually, this highlights the fact that is not clear which of the arguments are applied for both X and y and which one aren't. E.g, `copy` now applies to both X and y, but not `dtype` (or others). Dont know what is the best way to solve this without breaking the API... In any case, the exact behaviour of each argument should be properly documented. cc @jnothman @arjoly @amueller